### DISCUSSION TITLE
Changing Popup background overlay opacity

### DISCUSSION THREAD

**Discussion by BeepBeepBopBop** [CommunityToolkit/Maui#2291] [Q&A]
Hello, I would like to change the background overlay on a Popup.

I took a look at the code and found in PopUpHandler.windows.cs:

```
	static void AddOverlayToWindow(IWindow window)
	{
		var uiSetting = new UISettings();
		var backgroundColor = uiSetting.GetColorValue(UIColorType.Background).ToColor();
		window.AddOverlay(new PopupOverlay(window, backgroundColor.IsDark()
													? Color.FromRgba(0, 0, 0, 153)
													: Color.FromRgba(255, 255, 255, 153))); // 60% Opacity
	}

```

It seems I could achieve what I want by providing the wanted color to this method but I am not sure what would be the entry point using Handlers probably ?

Thanks for any help with that

**Comment by FireController1847** [comment]
+1 on this.

**Comment by matmork** [comment]
Would love this as well. Here are some more places I suspect the overlays are set.

### Mac/iOS
https://github.com/CommunityToolkit/Maui/blob/33bda603e922cb5b07435bcbb60dea5c235cda00/src/CommunityToolkit.Maui.Core/Views/Popup/MauiPopup.macios.cs#L168

### Windows
https://github.com/CommunityToolkit/Maui/blob/33bda603e922cb5b07435bcbb60dea5c235cda00/src/CommunityToolkit.Maui.Core/Views/Popup/PopupOverlay.windows.cs#L27
and/or
https://github.com/CommunityToolkit/Maui/blob/33bda603e922cb5b07435bcbb60dea5c235cda00/src/CommunityToolkit.Maui.Core/Handlers/Popup/PopUpHandler.windows.cs#L153

### Tizen
https://github.com/CommunityToolkit/Maui/blob/33bda603e922cb5b07435bcbb60dea5c235cda00/src/CommunityToolkit.Maui.Core/Views/Popup/MauiPopup.tizen.cs#L61

### Android
Guess something has to be overridden/set from the extended `Dialog`
https://github.com/CommunityToolkit/Maui/blob/33bda603e922cb5b07435bcbb60dea5c235cda00/src/CommunityToolkit.Maui.Core/Views/Popup/MauiPopup.android.cs#L12
